modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationParameters.java

```diff
@@ -110,6 +115,27 @@ private CalculationParameters of(Map<Class<? extends CalculationParameter>, Calc
     return new CalculationParameters(map);
   }
 
+  @ImmutableConstructor
+  private CalculationParameters(Map<Class<? extends CalculationParameter>, CalculationParameter> parameters) {
+    JodaBeanUtils.notNull(parameters, "parameters");
+    this.parameters = ImmutableMap.copyOf(parameters);
+    // find parameters that are super-interfaces of the specified objects
+    ImmutableMap.Builder<Class<? extends CalculationParameter>, Class<? extends CalculationParameter>> aliases =
+        ImmutableMap.builder();
+    for (Class<? extends CalculationParameter> type : parameters.keySet()) {
+      Class<?>[] interfaces = type.getInterfaces();
+      for (Class<?> iface : interfaces) {
+        if (iface != CalculationParameter.class &&
+            CalculationParameter.class.isAssignableFrom(iface) &&
+            !parameters.containsKey(iface)) {
+
+          aliases.put(iface.asSubclass(CalculationParameter.class), type);
```

Comment:
This will fail if there are two values in `parameters` of different classes that both implement the same interface that is a subtype of `CalculationParameter`. It's ambiguous which one should be used so it should probably fail, but it might be nice to provide some explanation.